pipfile_sort/__init__.py

```diff
@@ -1,30 +1,56 @@
-from click import command
-from click import option
-from click import version_option
+import sys
+from collections import defaultdict
+
+from click import command, option, version_option
 from plette import Pipfile
 from plette.pipfiles import PackageCollection
-import sys
 
-APP_VERSION = '0.2.2'
-PIPFILE_FILENAME = './Pipfile'
-PIPFILE_ENCODING = 'utf-8'
+APP_VERSION = "0.4.0"
+PIPFILE_FILENAME = "./Pipfile"
+PIPFILE_ENCODING = "utf-8"
 
 
 @command()
 @version_option(version=APP_VERSION)
-@option('--exit-code', is_flag=True, help=
-    'change to behavior of exit code. default behavior of return value, 0 is no differences, 1 is error exit. '
-    'return 2 when add this option. 2 is exists differences.')
-def main(exit_code):
+@option(
+    "--exit-code",
+    is_flag=True,
+    help="""Change behaviour of exit code:
+    - 0 is no differences
+    - 1 is error exit
+    - 2 is difference exists (added by this flag)
+    """,
+)
+@option(
+    "--case-sensitive",
+    is_flag=True,
+    help="If we want to sort with respect to case. If enabled, "
+    "uppercase & lowercase will be grouped together respectively",
+)
+@option(
+    "--no-group-types",
+    is_flag=True,
+    help="If we want to disable sorting according to datatype. If enabled, "
+    "semgrep, git will be grouped together respectively.",
+)
```

Comment:
Here's some unsolicited review from not-the-maintainer:

In my codebase we have some dependencies installed with extras like this:
```
...
Django = ">=4.2.0"
django-axes = { extras = ["ipware"], version = ">=6.0.1,<7" }
...
```

and I don't think it makes sense to move that to the same grouping as dependencies installed from git.

I think this flag would make more sense if it was inverted, i.e. you would do `pipfile-sort --group-by-type` or something if you wanted the grouping but not use it by default.

Could also try to find a better way to distinguish between git-installed deps and deps brackets because of extras and such. I do think there's a usecase for grouping those by themselves.